### PR TITLE
Fixing pretty printer to make its output readable by SANY

### LINKED ISSUE #574: [FEATURE] Rename variables with `$` when expoting modules to files
Internally, Apalache variable names contain `$` characters, due to renaming. Consequently, exporting internal representation as a `.tla` file produces files, which are not parseable, since variable names in TLA+  may not contain these characters (see #555).

We should solve this in the following way:
1. Introduce reserved variable names, e.g. those starting with an `APA_` prefix.
2. Implement a language watchdog pass immediately after parsing, which rejects specifications in which users introduce variable names belonging to the Apalache-reserved namespace (with a user-friendly warning)
3. Whenever variables are written to a file, they are stripped of illegal characters and given the Apalache-reserved prefix, to avoid the possibility of name collisions.

### LINKED ISSUE #785: [FEATURE] Preprocessed TLA+ files should have the same name as the module names
To be able to feed the files back to SANY.

### PR DESCRIPTION
**Pull request by konnov**
This PR fixes two issues:

- [x] closes #574: replacing the unsupported characters `$` and `!` with strings `_si_` and `_i_`.
- [x] closes #785: naming output files so they have the same name as the module name, in order to make it parseable by SANY.

The PR also outputs JSON in all passes.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: test/tla/cli-integration-tests.md [hunk 1]
```diff
@@ -164,6 +164,24 @@ EXITCODE: OK
...
+$ apalache-mc parse --output=output.tla Annotations.tla | sed 's/I@.*//'
+...
+EXITCODE: OK
+...
+```
+
+### parse --output=annotations.json Annotations succeeds
+
+```sh
+$ apalache-mc parse --output=output.json Annotations.tla | sed 's/I@.*//'
```
Comment (shonfeder) [inline test/tla/cli-integration-tests.md:179]: Any need/desire to also check that the output.json contains the expected content?

**Review by shonfeder — APPROVED** [review]
LGTM

File: test/tla/cli-integration-tests.md (hunk 1 above)
Comment (konnov) [reply test/tla/cli-integration-tests.md:179]: Yep. When Jure's PR is there, we will feed the output back to the reader ;-)

**Review by Kukovec — APPROVED** [review]
LGTM.

**Pull request merged**